src/filter/mod.rs

```diff
@@ -555,6 +576,7 @@ fn unapply2<'a>(
 ) -> JoshResult<git2::Tree<'a>> {
     return match op {
         Op::Nop => Ok(tree),
+        Op::Linear => Ok(tree),
```

Comment:
I can see that it works, but I don't understand why. How does it find the parents after this?

Reply:
Magic 🪄 

Just like with any other filter: When we need to find the original commits we search them by traversing the commit graph until we find `apply(sha) == filtered_sha`

Reply:
That makes sense. Thinking about it more, this didn't confuse me on other filters, but for some reason I got stuck on this one :)